### PR TITLE
check and set trigger time

### PR DESCRIPTION
**Pull request by sahiljhawar**
Solves #256, incomplete
@tylerbarna Can you have a look at this?

Do not merge!

### CODE REVIEW

File: nmma/em/analysis.py [hunk 1]
```diff
@@ -540,16 +540,27 @@ def analysis(args):
...
                 data = json.load(f)
                 for key in data.keys():
                     data[key] = np.array(data[key])
 
         if args.trigger_time is None:
-            raise ValueError("trigger_time required if using a data file.")
-
-        trigger_time = args.trigger_time
+            print("trigger_time is not provided, analysis will continue using the time of first data point") #maybe change it to warning instead of print
+            trigger_time=min_time #For now this will only work if data file is not in JSON or try block is executed successfully.
```
Comment (tylerbarna) [inline nmma/em/analysis.py:559]: We should probably support json lightcurves, at least in the format that nmma itself creates from injections

**Review by tylerbarna — CHANGES_REQUESTED** [review]

File: nmma/em/analysis.py (hunk 1 above)
Comment (tylerbarna) [reply nmma/em/analysis.py:559]: But yeah, this seems to be in line with what I was thinking for handling when a trigger time wasn't provided

Comment (sahiljhawar) [reply nmma/em/analysis.py:559]: @tylerbarna Do you have any json LC for test?

Comment (tylerbarna) [reply nmma/em/analysis.py:559]: > @tylerbarna Do you have any json LC for test?

@sahiljhawar [here's one](https://github.com/tylerbarna/dsmma_kn_23/blob/model-recovery/injections/lc_Me2017_00000.json)

File: nmma/em/analysis.py [hunk 2]
```diff
@@ -540,16 +540,27 @@ def analysis(args):
...
             with open(args.data) as f:
                 data = json.load(f)
                 for key in data.keys():
                     data[key] = np.array(data[key])
 
         if args.trigger_time is None:
-            raise ValueError("trigger_time required if using a data file.")
-
-        trigger_time = args.trigger_time
+            print("trigger_time is not provided, analysis will continue using the time of first data point") #maybe change it to warning instead of print
```
Comment (tylerbarna) [inline nmma/em/analysis.py:558]:
```suggestion

```

File: nmma/em/analysis.py [hunk 3]
```diff
@@ -540,16 +540,27 @@ def analysis(args):
...
+            trigger_time=min_time #For now this will only work if data file is not in JSON or try block is executed successfully.
+            print(f"trigger_time = {trigger_time}")
```
Comment (tylerbarna) [inline nmma/em/analysis.py:560]:
```suggestion
            print(f"trigger_time is not provided, analysis will continue using a trigger time of {trigger_time}") 
```

File: nmma/em/analysis.py (hunk 2 above)
Comment (sahiljhawar) [reply nmma/em/analysis.py:558]: I will include the edits before merging

**Review by mcoughlin — APPROVED** [review]
LGTM

**Pull request merged**